### PR TITLE
Legion/montgomp/extension client

### PR DESCRIPTION
**Pull request by montgomp**
image

### Testing 🧪
 > [ Add any special instructions needed to test or view your changes such as environment URLs or other config details. ]

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

**Comment by montgomp** [comment]
We'll probably want to rollback the ESLint things but doing so will not let prod builds occur

File: .eslintrc.prod.json [hunk 1]
```diff
@@ -1,16 +1,3 @@
...
-        "no-multiple-empty-lines": "error",
-        "react-hooks/exhaustive-deps": "off",
-        "@typescript-eslint/no-unused-vars": [
-            "error",
-            {
-                "argsIgnorePattern": "^_"
-            }
-        ]
-    }
+    "extends": "./.eslintrc.json"
```
[the remark this answers is not in the document]
Comment (msnyder-msft) [reply .eslintrc.prod.json:2]: I see your comment now.

File: src/Apps/Legion/EntryPoints/LegionApp/LegionApp.types.ts [hunk 2]
```diff
@@ -1,2 +1,134 @@
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface ILegionAppProps {}
+export interface ILegionAppProps {
+    extensionClient?: IExtensionClient;
+}
+
+export interface IExtensionClient {
```
[the remark this answers is not in the document]
Comment (msnyder-msft) [reply src/Apps/Legion/EntryPoints/LegionApp/LegionApp.types.ts:5]: Maybe something like ExternalTypes or ShellTypes or something of that nature?

File: .eslintrc.prod.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (montgomp) [reply .eslintrc.prod.json:2]: Yep. There's just some issues currently that won't compile into the tarball with the linting rules on. Will reveret and fix the issues but wanted to get everyone a chance to put eyes on this.

File: src/Apps/Legion/EntryPoints/LegionApp/LegionApp.types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (montgomp) [reply src/Apps/Legion/EntryPoints/LegionApp/LegionApp.types.ts:5]: Absolutely

File: .eslintrc.prod.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (montgomp) [reply .eslintrc.prod.json]: Reenabled. It seems the ongoing dev in main has resolved what I was seeing when I originally branched off.

**Review by pasanchMSFT — APPROVED** [review]
Looks good to me

**Pull request merged**